[BUG] kyverno-pre panic when checking kubernetes version

**Software version numbers**
State the version numbers of applications involved in the bug.

* Kubernetes version: v1.21.3-mi-v1.21.3-dev
* Kubernetes platform (if applicable; ex., EKS, GKE, OpenShift): selft-build
* Kyverno version: 1.5.1

**Describe the bug**
A clear and concise description of what the bug is.

When deploying kyverno in a self-buildt k8s cluster, kyverno-pre panic with the following errors:

```
panic: key is not a string: {"major":"1","minor":"21+","gitVersion":"v1.21.3-mi-v1.21.3-dev","gitCommit":"792f4b304ca623fca223d6f38c60fd94cb0c7c8f","gitTreeState":"clean","buildDate":"2021-08-23T05:44:33Z","goVersion":"go1.16.7","compiler":"gc","platform":"linux/amd64"}
goroutine 1 [running]:
k8s.io/klog/v2/klogr.flatten({0xc000502180, 0x2, 0x100000000})
	/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klogr/klogr.go:158 +0x4ff
k8s.io/klog/v2/klogr.klogger.Error({0x0, 0x0, {0x15578a4, 0x18}, {0x220b600, 0x0, 0x0}, {0x154980a, 0x9}}, {0x16edcc0, ...}, ...)
	/go/pkg/mod/k8s.io/klog/v2@v2.10.0/klogr/klogr.go:228 +0x573
github.com/kyverno/kyverno/pkg/utils.HigherThanKubernetesVersion(0xc000480fa0, {0x172af20, 0xc0003f6e60}, 0xc0003f6e60, 0xc0003f6e60, 0x15825d0)
	/kyverno/pkg/utils/util.go:163 +0x15d
main.main()
	/kyverno/cmd/initContainer/main.go:84 +0x36e
```
The reason is that the k8s version pattern is a liitle too strict:  match v(\d+).(\d+).(\d+)\.* only once and wrong usage of log.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/pkg/utils/util.go
+++ b/pkg/utils/util.go
@@ -160,7 +160,7 @@ func HigherThanKubernetesVersion(client *client.Client, log logr.Logger, major, 
 
 	b, err := isVersionHigher(serverVersion.String(), major, minor, patch)
 	if err != nil {
-		logger.Error(err, "serverVersion", serverVersion)
+		logger.Error(err, "serverVersion", serverVersion.String())
 		return false
 	}
 
@@ -168,27 +168,29 @@ func HigherThanKubernetesVersion(client *client.Client, log logr.Logger, major, 
 }
 
 func isVersionHigher(version string, major int, minor int, patch int) (bool, error) {
-	groups := regexVersion.FindAllStringSubmatch(version, -1)
-	if len(groups) != 1 || len(groups[0]) != 4 {
+	groups := regexVersion.FindStringSubmatch(version)
+	if len(groups) != 4 {
 		return false, fmt.Errorf("invalid version %s. Expected {major}.{minor}.{patch}", version)
 	}
 
-	currentMajor, err := strconv.Atoi(groups[0][1])
+	currentMajor, err := strconv.Atoi(groups[1])
 	if err != nil {
 		return false, fmt.Errorf("failed to extract major version from %s", version)
 	}
 
-	currentMinor, err := strconv.Atoi(groups[0][2])
+	currentMinor, err := strconv.Atoi(groups[2])
 	if err != nil {
 		return false, fmt.Errorf("failed to extract minor version from %s", version)
 	}
 
-	currentPatch, err := strconv.Atoi(groups[0][3])
+	currentPatch, err := strconv.Atoi(groups[3])
 	if err != nil {
 		return false, fmt.Errorf("failed to extract minor version from %s", version)
 	}
 
-	if currentMajor <= major && currentMinor <= minor && currentPatch <= patch {
+	if currentMajor < major ||
+		(currentMajor == major && currentMinor < minor) ||
+		(currentMajor == major && currentMinor == minor && currentPatch <= patch) {
 		return false, nil
 	}
 

--- a/pkg/utils/util_test.go
+++ b/pkg/utils/util_test.go
@@ -91,6 +91,15 @@ func Test_higherVersion(t *testing.T) {
 
 	v, err = isVersionHigher("v1.5.9-rc2", 1, 5, 9)
 	assert.Assert(t, v == false && err == nil)
+
+	v, err = isVersionHigher("v1.5.9", 2, 1, 0)
+	assert.Assert(t, v == false && err == nil)
+
+	v, err = isVersionHigher("v2.1.0", 1, 5, 9)
+	assert.Assert(t, v == true && err == nil)
+
+	v, err = isVersionHigher("v1.5.9-x-v1.5.9.x", 1, 5, 8)
+	assert.Assert(t, v == true && err == nil)
 }
 
 func Test_ConvertResource(t *testing.T) {

